### ISSUE TITLE
Analytics KPI dashboard request for Search & Discovery Team- Onsite Search KPIs

### ISSUE THREAD

**Issue by denisecoveyduc** [department-of-veterans-affairs/va.gov-team#16778]
# Request a KPI Dashboard

## What this form is for
Use this template to request a KPI dashboard on Domo for your product.

## Instructions for the VFS team
- [x] For additional info, read through the [How to Request a KPI Dashboard Documentation](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/analytics/kpi-dashboard-request.md)
- [x] Your KPIs are clearly linked back to Veteran outcomes
- [x] Link your KPIs below (KPIs should be in your Product Outline)
- [x] Add your team label to follow progress

Is this a new KPI dashboard request? Or is this an iteration? Please check the correct response.
- [x] New dashboard. 
- [ ] Iteration of current dashboard

### KPIs

|**KPI**         |**KPI Description**         |**Narrative**| **Display Preference**|
|-----------------------|------------------------------|------------------------------|------------------------------|
|Total Queries             |# of searches conducted |Captures difference in overall query numbers over time. |Assuming because of the adjustable date range this will be a line chart |
|Search Exits (Bounce Rate)             |% of queries that lead users to exit the site without selecting a result. |Percentage of searches overall where a user leaves the search results page without clicking on a link |Assuming because of the adjustable date range this will be a line chart |
|Refinement Rate             |% searches with query refinement |Percentage of searches overall where a user rephrases their search term |Assuming because of the adjustable date range this will be a line chart |
|Conversion Rate for search users             |# of users who conducted a query and were able to proceed to a VA page after selecting from the results page |How many users were successful in moving from "point A" (conducing a search) to "Point B".. (clicking on one of the search results page). |Assuming because of the adjustable date range this will be a line chart |
|Average Search Depth             |The number of pages visitors viewed after getting results for the search term. |Measures health of search result relevancy. More pages viewed means users need to look harder to find what they were looking for. |Assuming because of the adjustable date range this will be a line chart |

*Please leave the following blank*

---

## VSP Analytics & Insights Acceptance Criteria

1. Groom Analytics
- [x] Groom analytics for technical feasibility
    - [x] Document data sources for each metric
    - [ ] Document unavailable vs. available metrics
- [ ] If any, add implementation issue for additional necessary implementation

2. Draft PST Dashboard
- [ ] Draft dashboard on Domo
- [ ] Structure data for ETL
- [ ] Pipe available metrics to Domo
- [ ] Add appropriate team members to dashboard

Please see [checklist for full KPI dashboarding QA checklist](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/vsp/teams/insights-analytics/analytics-tools/domo-dashboard-qa-checklist.md)

## Definition of Done
- [ ] KPIs have been groomed and connected to Domo
- [ ] Dashboard has been shared back with the VFS team
- [ ] KPI Dashboard full QA checklist completed

**Comment by joanneesteban** [comment]
Thank you for the request, @denisecoveyduc ! Keep us posted on when you get your KPIs added, or if you would like a KPI setting workshop with us.

**Comment by joanneesteban** [comment]
@denisecoveyduc can you give us an update on when you're hoping to have your KPIs set or a KPI setting workshop?

**Comment by joanneesteban** [comment]
Hi @denisecoveyduc - checking in again, do you have an update on your KPIs? Or would you like to schedule a KPI setting workshop?

**Comment by denisecoveyduc** [comment]
@joanneesteban Thanks for reaching out! I have to apologize I wasn't receiving notifications for the first two times you messaged me. I have fixed that problem now though. We have actually been discussing this this week and today I am pulling together a summary of what we have identified with hopefully some thoughts on how we might want to see that visualized. I will post that once it is ready, you can review and then we can go from there and discuss in more detail.   Sound good? Thanks for your patience.

**Comment by joanneesteban** [comment]
That sounds great! Thanks, @denisecoveyduc !

**Comment by denisecoveyduc** [comment]
### Search & Discovery Team KPIs
|**KPI**         |**KPI Description**         |**Significance  for SD Team (1:High - 3:Low)**| **Narrative **|
|-----------------------|------------------------------|------------------------------|------------------------------|
|Customer Satisfaction 1             |Question 1: Did you find what you were looking for today? [‘yes’ or ‘no’] |1 |Defines user perception of their own success in finding what they were looking for |
|Customer Satisfaction 2               |Question 2: What were you unable to find during your visit today? [text area] |1 |Captures search terms that users were not able to find results for and provides examples of user phrasing |
|Customer Satisfaction 3             |Question 3: What is your overall satisfaction with this site? |1 |How content they users with the search experience on va.gov |
|Total Queries             |#of searches conducted |1 |Captures difference in overall query numbers over time. |
|Search Usage             |Percentage of va.gov visitors who conducted a search (both global search and mini searches) |2 |What is the increase a result of? |
|Unique Searches Per Session             |Avg. # search queries per session |3 |How many searches a single user conducts during one visit to va.gov |
|Zero Result Searches             |#of queries resulting in zero results |1 |number of searches that produce zero search results and the terms that produce zero results  |
|Search Exits (Bounce Rate) |% of queries that lead users to exit the site without selecting a result.(both global search and mini searches) |1 |Percentage of searches overall where a user leaves the search results page without clicking on a link |
|Refinement Rate             |% searches with query refinement |1 |Percentage of searches overall where a user rephrases their search term |
|Average Search Depth             |The number of pages visitors viewed after getting results for the search term. |2 |Measures health of search result relevancy. More pages viewed means users need to look harder to find what they were looking for. |
|Click Through Rate            |Click thru Rate from "conduct query" to search result page |1 |Number of clicks between conducing a search and selecting on of the results page. |
|Conversion Rate             |#of users who conducted a query and were able to proceed to a VA page after selecting from the results page |1 |How many users were successful in moving from "point A" (conducing a search) to "Point B".. (clicking on one of the search results page). |
Result Page Load Time| Length of time to load the results page|2|Length of time users spend waiting for search results page to load.|

@jonwehausen @mgleo and I have started to think about how we may like to have this data presented on the dashboard but we have a few questions. Should we work directly with @JasonTheBearded  or set up a workshop as @joanneesteban suggested?

**Comment by joanneesteban** [comment]
@denisecoveyduc thank for meeting with the team! Let me know if you want to set up a KPI setting workshop to help prioritize these.

**Comment by denisecoveyduc** [comment]
@jonwehausen @JasonTheBearded @joanneesteban  As discussed, I have update the KPIs above to reflect our top 5 for this MVP Dashboard.  Please review and let us know if further clarity is needed. The remaining KPIs we will consider for future additions to the dashboard. 

Also sharing the search.gov endpoint below which now captures click through traffic which is equivalent to the way we have described conversion rate above. 

************************************************
### Search.gov API endpoint URL

Information is under Getting Started on https://open.gsa.gov/api/searchgov-results/:
The endpoint is https://api.gsa.gov/technology/searchgov/v2/results/i14y
You must use https. You can find your access key on the API Access Key page of the Search.gov Admin Center.
************************************************

cc: @johnhashva @mgleo @srsuddath

**Comment by denisecoveyduc** [comment]
@joanneesteban Hi Joanne, with the updated request highlighting our top 5 KPI Metrics are you able to offer an estimated completion date so I can set expectations appropriately?

**Comment by denisecoveyduc** [comment]
@joanneesteban Any updates on timeline to establish this dashboard? Thanks ;)

**Comment by joanneesteban** [comment]
Hi @denisecoveyduc - we're a bit pressed for resources at the moment - we're anticipating being able to pull this into another sprint (around March). 

In the meantime, are you able to find some of these metrics in Google Analytics? We can help you find those links.

**Comment by denisecoveyduc** [comment]
Thanks @joanneesteban for the update. 

I have been able to find most of these metrics by using a combination of  GA and search.gov data but would love to validate that I am looking at the right information. I was planning to open a validation ticket to confirm I am looking at the right metrics or would you prefer I just add that here?

The only one that is missing is the "Conversion Rate for search users" which we discussed on our call this morning. If we are able to prioritize this metric that would be incredible as we would like to add that data to the typeahead dashboard as well.

**Comment by joanneesteban** [comment]
Thanks, @denisecoveyduc ! Sure, apologies about the overhead here: would you actually be able to open another ticket? That way we don't miss the dashboard part of your request.

**Comment by denisecoveyduc** [comment]
Hello to our favorite analytics team! Thank you all for your hard work. 💪

@joanneesteban Your plates are full enough, I have what I need for now and won't create the validation issue discussed above. 

However, to establish our initial **Onsite Search DOMO Dashboard** though we thought a quick visualization (created by @mgleo ) may make the process easier by highlighting what we are thinking. Let us know if you have any questions at all. 

We are envisioning that data from search.gov will be used to establish our conversion rate since they are now tracking click through rates.  Adding the search.gov API information for your reference.

**Search.gov APIs**
https://open.gsa.gov/api/searchgov-results/ - Main
https://open.gsa.gov/api/searchgov-clicks/ - Click tracking
https://open.gsa.gov/api/searchgov-suggestions/ - Typeahead 

Please let us know if you have any questions or need anything! Hang in there!
![Onsite KPI Dashboard Visualization](https://user-images.githubusercontent.com/72393866/110676244-c03ef000-81a1-11eb-866a-7cb27397db61.jpg)


cc: @johnhashva @mgleo @srsuddath

**Comment by denisecoveyduc** [comment]
@joanneesteban Curious if you are able to offer any estimated delivery time for this dashboard. We know you are quite busy, just want to get a rough sense of timing or where we are in the queue.  TIA

**Comment by denisecoveyduc** [comment]
Update on timing estimate based on Slack discussion with Joanne:
We’re estimating the end of June/beginning of July for delivering KPI Dashboards. We’ll likely use the standardized dashboards that you saw during the feedback session for this request.”
